compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -468,13 +468,8 @@ impl Attribute {
                 .all(|ch| {
                     ch.is_ascii_alphabetic()
                         || ch.is_numeric()
-                        || ch == '_'
-                        || ch == '('
-                        || ch == ')'
-                        || ch == '='
-                        || ch == '"'
+                        || ch.is_ascii_punctuation()
                         || ch == ' '
-                        || ch == '\''
```

Comment:
opted to simplify down since [the method covers most](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_punctuation) of the common cases we'd probs care about
 
i've done v little work on programming langs before, so if the explicit-ness of the manual list is preferred i'm happy to change it back

maybe to cover `~!@#$%^&*().,?';:"` or a similar subset

Reply:
Thats a goos shout -- I'd also prefer that method over manually specifying all of the cases we care about